### ISSUE TITLE
Fails to compile in typescript

### ISSUE THREAD

**Issue by BryanCrotaz** [ember-lifeline/ember-lifeline#167]
Addon uses `ember-cli-typescript` and `ember-lifeline`. When building its dummy app or during the `precompile` step of `npm publish` I get the following errors:

```
node_modules/ember-lifeline/dom-event-listeners.d.ts(45,114): error TS2304: Cannot find name 'RunMethod'.

node_modules/ember-lifeline/dom-event-listeners.d.ts(53,117): error TS2304: Cannot find name 'RunMethod'.

node_modules/ember-lifeline/run-task.d.ts(11,78): error TS2304: Cannot find name 'EmberRunTimer'.

node_modules/ember-lifeline/run-task.d.ts(45,94): error TS2304: Cannot find name 'EmberRunTimer'.

node_modules/ember-lifeline/run-task.d.ts(80,67): error TS2304: Cannot find name 'EmberRunQueues'.

node_modules/ember-lifeline/run-task.d.ts(80,124): error TS2304: Cannot find name 'EmberRunTimer'.

node_modules/ember-lifeline/run-task.d.ts(112,90): error TS2304: Cannot find name 'EmberRunTimer'.

node_modules/ember-lifeline/run-task.d.ts(144,46): error TS2304: Cannot find name 'EmberRunTimer'.

node_modules/ember-lifeline/run-task.d.ts(145,64): error TS2304: Cannot find name 'EmberRunTimer'.
```

**Comment by BryanCrotaz** [comment]
tsconfig.json is
```
{
  "compilerOptions": {
    "target": "es2017",
    "allowJs": true,
    "moduleResolution": "node",
    "allowSyntheticDefaultImports": true,
    "noImplicitAny": false,
    "noImplicitThis": true,
    "alwaysStrict": true,
    "strictNullChecks": true,
    "strictPropertyInitialization": true,
    "noFallthroughCasesInSwitch": true,
    "noUnusedLocals": true,
    "noUnusedParameters": true,
    "noImplicitReturns": true,
    "noEmitOnError": true,
    "noEmit": true,
    "inlineSourceMap": true,
    "inlineSources": true,
    "baseUrl": ".",
    "module": "es6",
    "paths": {
      "dummy/tests/*": [
        "tests/*"
      ],
      "dummy/*": [
        "tests/dummy/app/*",
        "app/*"
      ],
      "ember-lifeline": [
        "addon"
      ],
      "ember-lifeline/*": [
        "addon/*"
      ],
      "*": [
        "types/*"
      ]
    }
  },
  "include": [
    "app",
    "addon",
    "tests",
    "types"
  ]
}

```

**Comment by rwjblue** [comment]
Should all of our `@types` dev dependencies be deps?

**Comment by BryanCrotaz** [comment]
I don't know... possibly...

**Comment by scalvert** [comment]
Ya I think for consumers of this addon in other addons, we need to add `@types` to dependencies. 

In the case of the types above, they're added to `types/ember.d.ts`. I'm not sure how to export them so they're available to consumers. 

Any ideas?

As an aside, I don't grok why those types, which are in `@types/ember`, are not available in my addon and I have to redefine them. Perhaps related?

**Comment by scalvert** [comment]
Should be fixed by #412
